admin/common_auth/views.py

```diff
@@ -21,6 +24,47 @@ def login(request):
     context = {'form': form}
     return render(request, 'login.html', context)
 
+
 def logout(request):
     logout_user(request)
     return redirect('auth:login')
+
+
+# Permission restriction based on staff (only staff can register new users). Use is_staff boolean?
+def register(request):
+    if request.user.is_staff:
+        if request.method == 'POST':
+            form = CustomUserRegistrationForm(request.POST)
+            if form.is_valid():
+                email = form.cleaned_data['email']
+                password = form.cleaned_data['password1']
+                new_user = MyUser.objects.create_user(email=email, password=password)
+                new_user.first_name = form.cleaned_data['first_name']
+                new_user.last_name = form.cleaned_data['last_name']
+                group_perms = form.cleaned_data['group_perms']
+
+                for group in group_perms:
+                    new_user.groups.add(group)
+
+                new_user.save()
+                reset_form = PasswordResetForm({'email': new_user.email}, request.POST)
+                assert reset_form.is_valid()
+                reset_form.save(
+                    subject_template_name='common_auth/emails/account_creation_subject.txt',
+                    email_template_name='common_auth/emails/invitation_email.html',
+                    request=request
+                )
+                messages.success(request, 'Registration successful')  # add email reference here
+                return redirect('auth:register')
+            else:
+                print(form.errors)
+                context = {'form': form}
+                return render(request, 'register.html', context, status=400)
+        else:
+            reg_form = CustomUserRegistrationForm()
+            context = {'form': reg_form}
+            return render(request, 'register.html', context)
+    else:
+        # Best way to inform user: pop-up alert or display message after redirect?
```

Comment:
Looks like we're implementing the second option, which is fine. We can remove the comment.